pkgs/development/ruby-modules/bundler-app/default.nix

```diff
@@ -68,7 +68,10 @@ in
     ${lib.optionalString installManpages ''
     for section in {1..9}; do
       mandir="$out/share/man/man$section"
-      find -L ${basicEnv}/${ruby.gemPath}/gems/${basicEnv.name} \( -wholename "*/man/*.$section" -o -wholename "*/man/man$section/*.$section" \) -print -execdir mkdir -p $mandir \; -execdir cp '{}' $mandir \;
+
+      # See: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/ruby-modules/gem/default.nix#L184-L200
+      # See: https://github.com/rubygems/rubygems/blob/7a7b234721c375874b7e22b1c5b14925b943f04e/bundler/lib/bundler.rb#L285-L291
+      find -L ${basicEnv}/${ruby.gemPath}/${lib.optionalString (basicEnv.gemType == "git" || basicEnv.gemType == "url") "bundler/"}gems/${basicEnv.name} \( -wholename "*/man/*.$section" -o -wholename "*/man/man$section/*.$section" \) -print -execdir mkdir -p $mandir \; -execdir cp '{}' $mandir \;
```

Comment:
`${basicEnv}/${ruby.gemPath}/gems/${basicEnv.name}`

IIUC this is meant to represent the install path of the gem, so that we can scan it for man pages. From your investigation, the issue arises because the install path of gem from a git source is different than the default. Would it be simpler to only make the conditional here directly?

Having to implicitly pass value down for every user of bundled-common (which is at least bundlerEnv) when only bundler-app needs it seems not ideal.

Are there other parts of ruby infrastructure that need/care about the gem's install_path? If so it might be worth having it be an attribute of the gem record representation.

I'm new to nix so take my 2c with a grain of salt

Reply:
> because the install path of gem from a git source is different than the default

For whatever the bundler is invoked the `bundler/` is prefixed, in this case nix delegates bundler for things coming from `:url` and `:git` on Gemfile.

> Having to implicitly pass value down for every user of bundled-common (which is at least bundlerEnv) when only bundler-app needs it seems not ideal.

I didn't understood this, can you please clarify the statement :eyes:

From a blind guess, you're talking about propagating `gems.${pname}.gemType`, if so I didn't find any other suitable option to check if the package was coming from git/url. If you think this can be improved be sure to leave a comment!

> Are there other parts of ruby infrastructure that need/care about the gem's install_path? If so it might be worth having it be an attribute of the gem record representation.

There might be edge case for git gem depending on another git gem, I haven't tested that, I'll let you know testing it tomorrow (since its midnight here) if there's any issue, tho there shouldn't be because only referenced gem's files should be searched for copying, not of its dependencies.

Reply:
> I didn't understood this, can you please clarify the statement 

Not sure if this is the right terminology in nix, but by inhering we are adding more variables/entries for every user of bundled-common. That is  I mean that we are adding variables to the scope of both bundler-env and bundler-app when only bundler-app needs it. Because the relationship is implicit it is less clear why is the value being inherited.

> if so I didn't find any other suitable option to check if the package was coming from git/url

Do we need to know where it was downloaded or where was it installed (ie. the install_path)?
Because we already inherit all the gems, would it be possible to use the basicEnv.name to look up the gem in the gems attribute set to find whether whether the source.type is either git or url? That way we only need to modify bundler-app. 

I am fairly new to nix so I don't know how to test the changes locally. Apologies for not accompanying the suggestions with working code.

That said, the current solution works and fixes a bug.